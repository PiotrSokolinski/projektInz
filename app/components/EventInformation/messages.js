/*
 * EventInformation Messages
 *
 * This contains all the text for the EventInformation component.
 */

import { defineMessages } from 'react-intl'

export const scope = 'app.components.EventInformation'

export default defineMessages({
  description: {
    id: `${scope}.description`,
    defaultMessage: 'Description',
  },
  invited: {
    id: `${scope}.invated`,
    defaultMessage: 'Invited members',
  },
  host: {
    id: `${scope}.host`,
    defaultMessage: '- host',
  },
})
