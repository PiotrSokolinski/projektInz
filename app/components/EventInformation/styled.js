import styled from 'styled-components'

export const Container = styled.div``

export const Date = styled.div`
  font-size: 17px;
  font-weight: bold;
`

export const Time = styled.div`
  font-size: 17px;
  font-weight: bold;
`

export const DateContainer = styled.div`
  display: flex;
  justify-content: space-evenly;
`

export const Description = styled.div`
  margin-bottom: 15px;
`

export const Title = styled.div`
  font-size: 15px;
  margin-bottom: 5px;
  font-weight: bold;
  ${props => props.center && 'text-align: center'};
`

export const InvitedMembersList = styled.div`
  display: flex;
  flex-direction: column;
`

export const InvitedMemberContainer = styled.div`
  display: flex;
  justify-content: center;
`

export const FullName = styled.div`
  margin-right: 15px;
  ${props => props.isAuthor && 'font-weight: bold;'};
`
