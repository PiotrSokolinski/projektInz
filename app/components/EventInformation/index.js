/**
 *
 * EventInformation
 *
 */

import React from 'react'
import dayjs from 'dayjs'
import map from 'lodash/map'
// import PropTypes from 'prop-types';

import UserAvatar from 'components/UserAvatar'

import { FormattedMessage } from 'react-intl'
import messages from './messages'
import * as Styled from './styled'

const mockMembers = [
  {
    firstName: 'John',
    lastName: 'Doe',
    avatarUrl: '',
  },
  {
    firstName: 'John',
    lastName: 'Doe',
    avatarUrl: '',
  },
]

const EventInformation = () => (
  <Styled.Container>
    <Styled.DateContainer>
      <Styled.Date>{dayjs().format('DD/MM/YYYY')}</Styled.Date>
      <Styled.Time>
        {dayjs().format('MM:hh')}-{dayjs().format('MM:hh')}
      </Styled.Time>
    </Styled.DateContainer>
    <Styled.Title>
      <FormattedMessage {...messages.description} />
    </Styled.Title>
    <Styled.Description>
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys
      standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to
      PageMaker including versions of Lorem Ipsum.
    </Styled.Description>
    <Styled.Title center>
      <FormattedMessage {...messages.invited} />
    </Styled.Title>
    <Styled.InvitedMembersList>
      {map(mockMembers, (member, index) => (
        <Styled.InvitedMemberContainer>
          <Styled.FullName key={`el-member-${index}`}>
            {member.firstName}
            &nbsp;
            {member.lastName}
          </Styled.FullName>
          <UserAvatar image={member.avatarUrl} size="tiny" />
        </Styled.InvitedMemberContainer>
      ))}
    </Styled.InvitedMembersList>
  </Styled.Container>
)

EventInformation.propTypes = {}

export default EventInformation
