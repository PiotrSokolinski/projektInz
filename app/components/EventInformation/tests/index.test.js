/**
 *
 * Tests for EventInformation
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'
import { IntlProvider } from 'react-intl'

import EventInformation from '../index'
import { DEFAULT_LOCALE } from '../../../i18n'

describe('<EventInformation />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <EventInformation />
      </IntlProvider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <EventInformation />
      </IntlProvider>,
    )
    expect(firstChild).toMatchSnapshot()
  })
})
