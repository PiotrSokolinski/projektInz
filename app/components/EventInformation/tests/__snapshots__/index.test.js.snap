// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventInformation /> Should render and match the snapshot 1`] = `
<div
  class="styled__Container-sc-1hllk44-0 hlplpO"
>
  <div
    class="styled__DateContainer-sc-1hllk44-3 egEexW"
  >
    <div
      class="styled__Date-sc-1hllk44-1 eYgBLJ"
    >
      26/10/2019
    </div>
    <div
      class="styled__Time-sc-1hllk44-2 dAegMl"
    >
      10:11
      -
      10:11
    </div>
  </div>
  <div
    class="styled__Title-sc-1hllk44-5 dNDBdO"
  >
    <span>
      Description
    </span>
  </div>
  <div
    class="styled__Description-sc-1hllk44-4 jrVeIJ"
  >
    Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to PageMaker including versions of Lorem Ipsum.
  </div>
  <div
    class="styled__Title-sc-1hllk44-5 jiYyLv"
  >
    <span>
      Invited members
    </span>
  </div>
  <div
    class="styled__InvitedMembersList-sc-1hllk44-6 juViIt"
  >
    <div
      class="styled__InvitedMemberContainer-sc-1hllk44-7 clBYPS"
    >
      <div
        class="styled__FullName-sc-1hllk44-8 jbQiHk"
      >
        John
         
        Doe
      </div>
      <div
        class="tiny styled__Container-vinm4t-0 gDIJTt"
      >
        <img
          alt="avatar"
          class="avatar-image"
          src="IMAGE_MOCK"
        />
      </div>
    </div>
    <div
      class="styled__InvitedMemberContainer-sc-1hllk44-7 clBYPS"
    >
      <div
        class="styled__FullName-sc-1hllk44-8 jbQiHk"
      >
        John
         
        Doe
      </div>
      <div
        class="tiny styled__Container-vinm4t-0 gDIJTt"
      >
        <img
          alt="avatar"
          class="avatar-image"
          src="IMAGE_MOCK"
        />
      </div>
    </div>
  </div>
</div>
`;
